integration_test/login_widget_test.dart

```diff
@@ -0,0 +1,84 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:integration_test/integration_test.dart';
+import 'package:survey/main.dart';
+
+void main() {
+  IntegrationTestWidgetsFlutterBinding.ensureInitialized();
+
+  group('Login test', () {
+    final binding = IntegrationTestWidgetsFlutterBinding.ensureInitialized()
+        as IntegrationTestWidgetsFlutterBinding;
+
+    binding.framePolicy = LiveTestWidgetsFlutterBindingFramePolicy.fullyLive;
+
+    testWidgets('When starting, it display login form correctly',
+        (tester) async {
+      await tester.pumpWidget(SurveyApp());
+
+      await binding.watchPerformance(() async {},
+          reportKey: 'animation_performance_summary');
+
+      final emailField = find.widgetWithText(TextFormField, 'Email');
+      final passwordField = find.widgetWithText(TextFormField, 'Password');
+      final loginButton = find.widgetWithText(ElevatedButton, 'Login');
+
+      expect(emailField, findsOneWidget);
+      expect(passwordField, findsOneWidget);
+      expect(loginButton, findsOneWidget);
+    });
+
+    testWidgets('When login with empty fields, it shows errors',
```

Comment:
according to your test case behavior, I think the description should be modified a bit
```suggestion
    testWidgets('When login with empty or invalid fields, it shows errors',
```

Reply:
@manh-t while it's true that it could be flaky, I think the most important for now is to setup the test to run first. To inject a different network layer to serve mock data and avoid flaky test could be done separately. Sometimes, in some cases like this, it doesn't take any extra step to validate the behavior, hence I make it directly. Later, cases like when we have to login first before fetching the Surveys, it will be then covered by the mock network